src/main/java/cmput402/tdd/App.java

```diff
@@ -158,6 +158,80 @@ public void editShoppingList(Scanner input, ShoppingList shoppingList){
         }
 
     }
+    public void editRecipe(Scanner input, Recipe recipe) throws Exception{
+        clearConsole();
+        System.out.println("Please select an option to edit your recipe:");
+        System.out.println("command format (ignore bracket): (add 1) (remove 5) (rename \"apples\")");
+        System.out.println("*add\n*remove\n*rename\n");
+        Integer quantity;
+        Item item;
+        String param;
+        String command = input.nextLine();
+        /*
+        Eg:
+        command = add 5
+
+        function will split command into variables action and param
+        action = add
+        param = 5
+         */
+        Integer spaceIndex = command.indexOf(" ");
+        if(spaceIndex == -1){
+            throw new Exception("Invalid command format");
+        }
+        String action = command.substring(0, spaceIndex);
+
+        if(action.equals("add") || action.equals("remove")) {
+            param = command.substring(spaceIndex + 1, command.length());
+            try{
+                quantity = Integer.parseInt(param);
+            }
+            catch(Exception e){
+                throw new Exception("Please enter a positive whole number for quantity");
+            }
+            try{
+                item = createItem(input);
+            }
+            catch (Exception e) {
+                throw new Exception("Failed to create item for recipe");
+            }
+            if(action.equals("add")){
+                if(quantity > 0){
+                    recipe.add(item, quantity);
+                    System.out.println("Added "+param+" "+item.getName()+" to recipe successfully");
+                }
+                else{
+                    throw new Exception("Could not add item to recipe");
+                }
+
+            }
+            else if(action.equals("remove")){
+                if(recipe.items.get(item) > quantity && quantity > 0){
+                    recipe.remove(item, quantity);
+                    System.out.println("Removed "+param+" "+item.getName()+" from recipe successfully");
+                }
+
+                if(recipe.items.get(item) == quantity) {
+                    recipe.remove(item);
+                }
+                else{
+                    throw new Exception("Could not remove item from recipe");
+                }
+            }
+        }
+        else if(action.equals("rename")){
+            param = command.substring(spaceIndex+2, command.length()-1);
+            if(param.length() <= 0 || param.length() > 40 || !action.equals("rename")){
```

Comment:
I don't understand why you need to check if the action does not equal "rename". If that were the case then we would not even be in this block

Reply:
Removed redundant condition in new commit